_docs/users.md

```diff
@@ -0,0 +1,28 @@
+# Users
+
+## Removing users
+
+Any user can request the deletion of their account. This can be done via user settings.
+
+### How is this working
+
+When a user deletes their account, it is soft-deleted. Along with it, all user's posts, 
+replies to her/his posts and created discussions are deleted in the same way.
```

Comment:
I think for replies to their posts, we should just change the parent of the next post? That allows discussions to remain intact, even though the single post was deleted. I've seen that happen on several other systems. What do you think? 

I think it makes sense to delete all posts for their discussions, of course.

Reply:
Well... this is tough. I took the current approach because there are many disadvantages when we leave replies to deleted posts.

* If we change the parent, the replies may be completely out of context of the other conversation's point of view
* Replies may interfere with the ongoing conversation (different publication times).
* What if there are no other parent posts? Should we make all replies top-level posts? We will face the same problems as above.
* There will be no way to undo this action when the user decides to recover their account.

IMO the ideal solution, when deleting a single post that has replies, would be:
* Not display the username and message of that post, but leave the replies so that the conversation structure can remain intact.
* This would make recovery easier, but would be difficult to handle during hard deletion... we would need an additional variable to indicate this special type of post (deleted, but not for real).

Reply:
I think there are relatively simple solutions to the issues you had, BUT - I rather like your ideal solution, honestly. I'm all for that if you want to do it.

Reply:
Ok, I will try to implement this.